lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -778,6 +779,126 @@ class DecomposeAtenConv2dOp : public OpRewritePattern<AtenConv2dOp> {
+  LogicalResult matchAndRewrite(AtenConvolutionBackwardOverrideableOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Value none = rewriter.create<ConstantNoneOp>(op->getLoc());
+    rewriter.replaceOpWithNewOp<AtenConvolutionBackwardOp>(
+        op, op.getResultTypes(), op.grad_output(), op.input(), op.weight(),
+        none, op.stride(), op.padding(), op.dilation(), op.transposed(),
+        op.output_padding(), op.groups(), op.output_mask());
+
+    return success();
+  }
+};
+} // namespace
+
+// Decompose aten.convolution_backward_overrideable to aten.convolution_backward
+namespace {
+class DecomposeAtenConvolutionBackwardOp
+    : public OpRewritePattern<AtenConvolutionBackwardOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenConvolutionBackwardOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+    MLIRContext *context = op.getContext();
+    Value ci0 = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(0));
+    Value ci1 = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    Value cNone = rewriter.create<Torch::ConstantNoneOp>(loc);
+    Value cFalse = rewriter.create<Torch::ConstantBoolOp>(
+        loc, rewriter.getBoolAttr(false));
+
+    Value gradOutput = op.grad_output();
+    Value input = op.input();
+    Value weight = op.weight();
+    auto gradRank = getTensorRank(gradOutput);
+    SmallVector<Value> padding;
+    getListConstructElements(op.padding(), padding);
+    SmallVector<int64_t> paddingInts;
+    if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts)))
+      return rewriter.notifyMatchFailure(op,
+                                         "only constant padding is supported");
+
+    if (gradRank != 4)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only 2D convolutions supported");
+
+    // Rotate weight
+    SmallVector<Value> axis;
+    for (auto i = 2; i < gradRank; i++)
+      axis.push_back(rewriter.create<Torch::ConstantIntOp>(
+          loc, rewriter.getI64IntegerAttr(i)));
+    Value axisList = rewriter.create<Torch::PrimListConstructOp>(
+        loc, Torch::ListType::get(Torch::IntType::get(op.getContext())), axis);
+    weight = rewriter.create<Torch::AtenFlipOp>(loc, weight.getType(), weight,
+                                                axisList);
+
+    // Transpose inputs
+    Value gradOutputTransposed = rewriter.create<Torch::AtenTransposeIntOp>(
+        loc, gradOutput.getType(), gradOutput, ci0, ci1);
+    Value inputTransposed = rewriter.create<Torch::AtenTransposeIntOp>(
+        loc, input.getType(), input, ci0, ci1);
+    Value weightTransposed = rewriter.create<Torch::AtenTransposeIntOp>(
+        loc, weight.getType(), weight, ci0, ci1);
+
+    // Convolve grad_output with weight (padding = [1, 1])
```

Comment:
I am not familiar with the algorithm. Is the padding always `1,1` for all cases here?

Reply:
Doing some further testing, it looks like it isn't. For stride=1 and dilation=1 it looks like it's the size of the weight - 1. For other values of stride and dilation I'm not sure what it's supposed to be. I'll research further in depth but for now I'm going to add an unsupported check for dilation>1 and stride>1 and put in weight size -1.

Reply:
And a comment and a // TODO here would be good.

Reply:
As per my comment below, I'm adding an explanation and keeping it as-is.

Reply:
You're right, I've double checked the math and I have what should be the correct implementation. I'm working on implementing it, but I'm running into issues with `tensor::PadOp`. I'm going to try to write it with a different op (`tensor::InsertSlice`) and push that up tomorrow.